tests/tree_depth_heuristics_tests.py

```diff
@@ -78,3 +78,23 @@ def test_prefer_shallow_depths(self):
         solutions = heuristics.prefer_shallow_depths(solutions, 0.5)
         self.assertEqual(solutions[0].weight, 1.0)
         self.assertTrue(solutions[1].weight < solutions[0].weight)
+
+    def test_prefer_no_markerless_sandwich(self):
+        """Generate two solutions, one in which a markerless sandwich
+        is used to skip depth, and another where it is not used to
+        skip depth."""
+        self.addAssignment(markers.ints, '1', 0)
+        self.addAssignment(markers.markerless, markers.MARKERLESS, 1)
+        self.addAssignment(markers.roman, 'i', 1)
+        solution1 = self.solution
+
+        self.setUp()
+        self.addAssignment(markers.ints, '1', 0)
+        self.addAssignment(markers.markerless, markers.MARKERLESS, 1)
+        self.addAssignment(markers.roman, 'i', 2)
+        solution2 = self.solution
+
+        solutions = [Solution(solution1), Solution(solution2)]
+        solutions = heuristics.prefer_no_markerless_sandwich(solutions, 0.5)
+        self.assertEqual(solutions[0].weight, 1.0)
+        self.assertTrue(solutions[1].weight < solutions[0].weight)
```

Comment:
Please also add a test along the lines of `[MARKERLESS, MARKERLESS, 'a', 'b', MARKERLESS, 'a', 'b']`?